var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -14,59 +14,67 @@ class Esmf(MakefilePackage):
 
-    homepage = "https://www.earthsystemcog.org/projects/esmf/"
+    homepage = "https://earthsystemmodeling.org/"
     url = "https://github.com/esmf-org/esmf/archive/ESMF_8_0_1.tar.gz"
     git = "https://github.com/esmf-org/esmf.git"
 
-    maintainers("climbfuji", "jedwards4b", "AlexanderRichert-NOAA")
+    maintainers("climbfuji", "jedwards4b", "AlexanderRichert-NOAA", "theurich", "uturuncoglu")
 
     # Develop is a special name for spack and is always considered the newest version
     version("develop", branch="develop")
     # generate chksum with spack checksum esmf@x.y.z
+    version("8.4.1", sha256="1b54cee91aacaa9df400bd284614cbb0257e175f6f3ec9977a2d991ed8aa1af6")
     version("8.4.0", sha256="28531810bf1ae78646cda6494a53d455d194400f19dccd13d6361871de42ed0f")
     version("8.3.1", sha256="6c39261e55dcdf9781cdfa344417b9606f7f961889d5ec626150f992f04f146d")
-    version("8.3.0", sha256="0ff43ede83d1ac6beabd3d5e2a646f7574174b28a48d1b9f2c318a054ba268fd")
-    version("8.3.0b09", commit="5b7e546c4b")
+    version(
+        "8.3.0",
+        sha256="0ff43ede83d1ac6beabd3d5e2a646f7574174b28a48d1b9f2c318a054ba268fd",
+        deprecated=True,
+    )
+    version("8.3.0b09", commit="5b7e546c4b", deprecated=True)
     version("8.2.0", sha256="3693987aba2c8ae8af67a0e222bea4099a48afe09b8d3d334106f9d7fc311485")
     version("8.1.1", sha256="58c2e739356f21a1b32673aa17a713d3c4af9d45d572f4ba9168c357d586dc75")
+    version(
+        "8.1.0",
+        sha256="367dc89ab8e273f6fbe3d63a64a9a6735cbaa77d5f144ebae4af198fee5bb203",
+        deprecated=True,
+    )
     version("8.0.1", sha256="9172fb73f3fe95c8188d889ee72fdadb4f978b1d969e1d8e401e8d106def1d84")
-    version("8.0.0", sha256="051dca45f9803d7e415c0ea146df15ce487fb55f0fce18ca61d96d4dba0c8774")
-    version("7.1.0r", sha256="ae9a5edb8d40ae97a35cbd4bd00b77061f995c77c43d36334dbb95c18b00a889")
+    version(
+        "8.0.0",
+        sha256="051dca45f9803d7e415c0ea146df15ce487fb55f0fce18ca61d96d4dba0c8774",
+        deprecated=True,
+    )
+    version(
+        "7.1.0r",
+        sha256="ae9a5edb8d40ae97a35cbd4bd00b77061f995c77c43d36334dbb95c18b00a889",
+        deprecated=True,
+    )
 
     variant("mpi", default=True, description="Build with MPI support")
-    variant("external-lapack", default=False, description="Build with external LAPACK support")
+    variant("external-lapack", default=False, description="Build with external LAPACK library")
     variant("netcdf", default=True, description="Build with NetCDF support")
     variant("pnetcdf", default=True, description="Build with pNetCDF support", when="+mpi")
-    variant("xerces", default=True, description="Build with Xerces support")
+    variant("xerces", default=False, description="Build with Xerces support")
     variant(
-        "parallelio",
+        "external-parallelio",
         default=True,
         description="Build with external parallelio library",
-        when="@8.3:",
+        when="@8.3.1:+mpi",
```

Comment:
Parallelio~mpi works fine, why was this changed to have the +mpi?

Reply:
It's actually not true - I build without mpi but include pio on the gust system.  Here is the spack spec:
```
esmf@8.5.0b17%intel@2021.7.1+debug+external-lapack~mpi+parallelio~xerces esmf_os=Unicos
    ^cray-libsci
    ^netcdf-c+dap~mpi
    ^parallelio@2.5.10~mpi~pnetcdf build_type=Debug
```

Reply:
@jedwards4b - can you point me to the resulting `esmf.mk` for this built? I am pretty sure it will have PIO turned OFF, but would like to verify.

Reply:
/glade/u/apps/cseg/spack/opt/spack/linux-sles15-zen3/intel-2021.7.1/esmf-8.4.1b02-mdffrgnov7swkh7osc7w74fg3c7ed4vn/lib/esmf.mk
```
#
# !!! The following options were used on this ESMF build !!!
#
# ESMF_DIR: /glade/gust/scratch/jedwards/spack-stage/spack-stage-esmf-8.4.1b02-mdffrgnov7swkh7osc7w74fg3c7ed4vn/spack-src
# ESMF_OS: Linux
# ESMF_MACHINE: x86_64
# ESMF_ABI: 64
# ESMF_COMPILER: intel
# ESMF_BOPT: O
# ESMF_COMM: mpiuni
# ESMF_SITE: default
# ESMF_PTHREADS: ON
# ESMF_OPENMP: ON
# ESMF_OPENACC: OFF
# ESMF_ARRAY_LITE: FALSE
# ESMF_NO_INTEGER_1_BYTE: TRUE
# ESMF_NO_INTEGER_2_BYTE: TRUE
# ESMF_FORTRANSYMBOLS: default
# ESMF_MAPPER_BUILD: OFF
# ESMF_AUTO_LIB_BUILD: ON
# ESMF_DEFER_LIB_BUILD: ON
# ESMF_SHARED_LIB_BUILD: ON
# 
# ESMF environment variables pointing to 3rd party software:
# ESMF_MOAB:              internal
# ESMF_LAPACK:            system
# ESMF_LAPACK_LIBS:       -lsci_intel
# ESMF_ACC_SOFTWARE_STACK:            none
# ESMF_NETCDF:            nc-config
# ESMF_NETCDF_INCLUDE:    /glade/work/csgteam/spack-deployments/gust/22.12/envs/build/opt/__spack_path_placeholder__/__spack_path_placeholder__/__spack_pa/netcdf/4.9.0/packages/netcdf-c/4.9.0/intel/2021.7.1/include
# ESMF_NETCDF_LIBS:       -lnetcdf
# ESMF_NETCDF_LIBPATH:    /glade/work/csgteam/spack-deployments/gust/22.12/envs/build/opt/__spack_path_placeholder__/__spack_path_placeholder__/__spack_pa/netcdf/4.9.0/packages/netcdf-c/4.9.0/intel/2021.7.1/lib
# ESMF_NFCONFIG:          nf-config
# ESMF_NETCDFF_INCLUDE:   /glade/u/apps/gust/22.12/spack/opt/spack/netcdf/4.9.0/packages/netcdf-fortran/4.6.0/intel/2021.7.1/include
# ESMF_NETCDFF_LIBS:      -lnetcdff -lnetcdf -lnetcdf
# ESMF_NETCDFF_LIBPATH:   /glade/u/apps/gust/22.12/spack/opt/spack/netcdf/4.9.0/packages/netcdf-fortran/4.6.0/intel/2021.7.1/lib
# ESMF_PIO:               external
# ESMF_PIO_INCLUDE:       /glade/u/apps/cseg/spack/opt/spack/linux-sles15-zen3/intel-2021.7.1/parallelio-2.5.10-o56vie7zy3jbqxshdcgg5p6rzksgd2by/include
# ESMF_PIO_LIBS:          -lpioc
# ESMF_PIO_LIBPATH:       /glade/u/apps/cseg/spack/opt/spack/linux-sles15-zen3/intel-2021.7.1/parallelio-2.5.10-o56vie7zy3jbqxshdcgg5p6rzksgd2by/lib
# ESMF_YAMLCPP:           internal
```

Reply:
Good. Have you built an executable with it, and it runs successfully?

Reply:
Yes, /glade/gust/scratch/jedwards/SMS_Mmpi-serial.f19_g17.X.gust_gnu.20230316_141050_qtdhib

Reply:
Well, that is great to know, and slightly unexpected. I did not think this would end so well because there would be a mix of mpi stub libraries going on. Therefore, there are MPI_ symbols built into the libesmf.a/.so, and I assumed that would conflict when linking (or maybe when loading at run-time) the serial-mpi that ParallelIO uses.

Apparently not a problem after all, at least not on Gust judging from your finding.

I will revert the change to keep supporting +external-parallelio~mpi.

Reply:
@jedwards4b One more question here. In your `esmf@8.5.0b17%intel@2021.7.1+debug+external-lapack~mpi+parallelio~xerces esmf_os=Unicos` example of the current behavior, I assume Spack found `^parallelio@2.5.10~mpi~pnetcdf build_type=Debug` because it was already there? If it was not there, it would have probably built `parallelio@2.5.10+mpi`? That is what I am seeing with the current Spack develop branch. I.e. the `~mpi` option given to `esmf` is not propagated down to `parallelio` automatically if it needs yet to be built. 

Do you confirm that, and do you see an issue with that?

Reply:
I confirm that - the spec had to be defined very explicitly - this is a feature or limitation of spack.

Reply:
So I think I have that working now (see https://github.com/spack/spack/commit/e82ed24a1fa8c3c6044c44db68d0a6fea7c8b46d), where the `+mpi/~mpi` variant is passed down to the `parallelio` dependency. I had some issue getting things build all the way through locally until I made a small change in my local `netcdf-c/package.py`, basically again handing the `+mpi/~mpi` through to `hdf5`, or else it was trying to pick a `+mpi` variant, but then failed linking against it during the `parallelio` install. But that's outside of ESMF, and I guess who ever is using `~mpi` with `+external-parallel` knows what they are doing.